back-end/apps/api/src/guards/frontend-version.guard.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  HttpException,
+  HttpStatus,
+  Injectable,
+  Logger,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import * as semver from 'semver';
+
+@Injectable()
+export class FrontendVersionGuard implements CanActivate {
+  private readonly logger = new Logger(FrontendVersionGuard.name);
+
+  constructor(private readonly configService: ConfigService) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const frontendVersion = request.headers['x-frontend-version'];
+    const minimumVersion = this.configService.get<string>('MINIMUM_SUPPORTED_FRONTEND_VERSION');
+    const clientIp = request.headers['x-forwarded-for'] || request.ip || 'unknown';
+
+    const UPGRADE_REQUIRED = 426;
+
+    if (!frontendVersion) {
+      this.logger.warn(`Request rejected: Missing x-frontend-version header from IP ${clientIp}`);
+      throw new HttpException(
+        {
+          statusCode: UPGRADE_REQUIRED,
+          message: 'Frontend version header is required. Please update your application.',
+          error: 'Upgrade Required',
```

Comment:
We should probably include the url, latestVersion, and minimumVersion values here as well, right? The idea being that if the app is rejected, they will then need the link to update.  So adding lines like
```
   downloadUrl: this.downloadUrl,
   latestVersion: this.latestVersion,
   minimumVersion: this.minimumVersion,
```
Or similar. Repeated for all exceptions.